pkg/compose/convergence.go

```diff
@@ -355,6 +355,9 @@ func shouldWaitForDependency(serviceName string, dependencyConfig types.ServiceD
 		// already managed by InDependencyOrder
 		return false, nil
 	}
+	if !dependencyConfig.Required {
```

Comment:
it SHOULD wait IF dependency has been detected. Should only ignore dependency was not found, but here you just bypass waiting for dependency whenever it's been detected or not